Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift

```diff
@@ -304,6 +307,12 @@ extension DefaultPublisher {
     }
 
     private func performAddOrTrack(_ trackable: Trackable, completion: Callback<Void>){
+        guard !duplicateTrackableGuard.isCurrentlyAddingTrackableWithId(trackable.id) else {
+            self.logHandler?.debug(message: "\(Self.self): Trackable with ID \(trackable.id) is already being added; saving completion handler.", error: nil)
+            duplicateTrackableGuard.saveDuplicateAddCompletionHandler(completion, forTrackableWithId: trackable.id)
```

Comment:
I know that's because I'm not familiar with this, but I can't find where `saveDuplicateAddCompletionHandler` is called to add the very first callback?

Reply:
We don't use the `DuplicateTrackableGuard` to save the callback for the _first_ `track` call for the trackable. It's only used for saving the completion handler for the subsequent calls (hence the `duplicate` in `saveDuplicateAddCompletionHandler`).